File: src/com/cattailsw/nanidroid/dlgs/ErrMsgDlg.java
package com.cattailsw.nanidroid.dlgs;

import com.cattailsw.nanidroid.R;
import com.cattailsw.nanidroid.R.string;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;

public class ErrMsgDlg extends DialogFragment {

    private static final String S_TITLE = "title";
    private static final String S_MSG = "msg";
private int tR;
private int mR;
    public static ErrMsgDlg newInstance(int titleRes, int msgRes) {
	ErrMsgDlg frag = new ErrMsgDlg();
        Bundle args = new Bundle();
        args.putInt(S_TITLE, titleRes);
	args.putInt(S_MSG, msgRes);
        frag.setArguments(args);
        frag.tR = titleRes;
        frag.mR = msgRes;
	return frag;
    }

	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState) {
	    /*int tR = savedInstanceState.getInt(S_TITLE);
	    int mR = savedInstanceState.getInt(S_MSG);
*/
		AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
		builder.setMessage(mR)
		.setTitle(tR)
		.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
			
			@Override
			public void onClick(DialogInterface dialog, int which) {
				// TODO Auto-generated method stub
				dialog.dismiss();
			}
		});
		
		return builder.create();
	}

}

File: src/com/cattailsw/nanidroid/dlgs/ErrMsgDlg.kt
package com.cattailsw.nanidroid.dlgs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment

class ErrMsgDlg : DialogFragment() {
    interface ErrDlgCallback { fun onDismiss(flag: Int) }
    private var tR = 0; private var mR = 0; private var flag = 0; private var cb: ErrDlgCallback? = null
    companion object {
        private const val S_TITLE = "title"; private const val S_MSG = "msg"
        @JvmStatic fun newInstance(titleRes: Int, msgRes: Int): ErrMsgDlg = newInstance(titleRes,msgRes,null,-1)
        @JvmStatic fun newInstance(titleRes: Int,msgRes: Int,cb: ErrDlgCallback?,flag: Int): ErrMsgDlg = ErrMsgDlg().also { f -> f.arguments=Bundle().apply{putInt(S_TITLE,titleRes);putInt(S_MSG,msgRes)};f.tR=titleRes;f.mR=msgRes;f.cb=cb;f.flag=flag }
    }
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog = AlertDialog.Builder(activity).setMessage(mR).setTitle(tR).setPositiveButton(android.R.string.ok) { dialog,_ -> dialog.dismiss();cb?.onDismiss(flag) }.create()
}
